src/qtcore/qml/QMLEngine.js

```diff
@@ -88,11 +82,18 @@ QMLEngine = function (element, options) {
       }
     }
 
+    this.extractBasePath = function( file ) {
+       var basePath = file.split(/[\/\\]/); // work both in url ("/") and windows ("\", from file://d:\test\) notation
+       basePath[basePath.length - 1] = "";
+       basePath = basePath.join("/");
+       return basePath;
+    }
     // Load file, parse and construct (.qml or .qml.js)
+    //TODO-MERGE check if this is merged correctly
```

Comment:
Or should we keep it in `master` until everything gets merged? What's the intention here?